py3status/modules/wwan_status_nm.py

```diff
@@ -0,0 +1,160 @@
+    format_down = 'WWAN: {state} - {operator} {netgen} ({signal}%)'
+    format_error = 'WWAN: {error}'
+    format_up = 'WWAN: {state} - {operator} {netgen} ({signal}%)'
+    modem = None
+
+    def wwan_status_nm(self):
+        response = {}
+        response['cached_until'] = self.py3.time_in(self.cache_timeout)
+
+        states = {10: "Connecting", 11: "Connected"}
+
+        # https://www.freedesktop.org/software/ModemManager/api/1.0.0/ModemManager-Flags-and-Enumerations.html#MMModemAccessTechnology
+        speed = {
+            16384: 'LTE',
+            8192: 'EVDOB',
+            4096: 'EVDOA',
+            2048: 'EVDO0',
+            1024: '1XRTT',
+            512: 'HSPA+',
+            256: 'HSPA',
+            128: 'HSUPA',
+            64: 'HSDPA',
+            32: 'UMTS',
+            16: 'EDGE',
+            8: 'GPRS',
+            4: 'GSM_COMPACT',
+            2: 'GSM',
+            0: 'POTS'
+        }
+
+        bus = SystemBus()
+        for id in range(0, 20):
+
+            # find the modem
+            device = 'Modem/' + str(id)
+
+            try:
+                proxy = bus.get('.ModemManager1', device)
+
+                # we can maybe choose another selector
+                eqid = str(proxy.EquipmentIdentifier)
+
+                if self.modem is None or self.modem == eqid:
+
+                    # get status informations
+                    status = proxy.GetStatus()
+
+                    # start to build return data dict
+                    data = {
+                        'state': 'Disconnected',
+                        'signal': status['signal-quality'][0],
+                        'netgen': speed[status['access-technologies']]
+                    }
+
+                    # if registred on network, get operator name
+                    if status['m3gpp-registration-state'] == 1:
+                        data['operator'] = status['m3gpp-operator-name']
+                    else:
+                        data['operator'] = STRING_UNKNOW_OPERATOR
+                        """
+                        break to be able to manage no/unplugged device
+                        we break here and not later to be able
+                        to keep information will disconnected
+                        """
+                        break
+
+                    # if connected or connecting
+                    if status['state'] == 11 or status['state'] == 10:
+                        data['state'] = states[status['state']]
+                        response['full_text'] = self.py3.safe_format(
+                            self.format_up, data)
+                        response['color'] = self.py3.COLOR_GOOD
+                    # else disconnected
+                    else:
+                        response['full_text'] = self.py3.safe_format(
+                            self.format_down, data)
+                        response['color'] = self.py3.COLOR_BAD
+                    return response
```

Comment:
For better readability, can we put everything in a dict later instead of doing it early?
```
full_text = 'some_text'
color = self.py3.COLOR_BAD
```
```
return {
    'full_text': full_text,
    'color': color,
}
```
Replace `return response` with `break` to get out of the loop immediately.